src/wasm.h

```diff
@@ -2255,11 +2255,14 @@ enum class ModuleItemKind {
 
 class Export {
 public:
-  // exported name - note that this is the key, as the internal name is
-  // non-unique (can have multiple exports for an internal, also over kinds)
+  // exported name - note that this is the key, as the internal name,
+  // or the exported type, is non-unique (can have multiple exports for an
+  // internal, also over kinds)
   Name name;
-  Name value; // internal name
+  std::variant<Name, HeapType> value; // internal name or exported type
   ExternalKind kind;
+  Name& getInternalName() { return std::get<Name>(value); }
+  bool hasInternalName() { return std::get_if<Name>(&value); }
```

Comment:
Elsewhere we combine these two by having the `getXXX` helper return a null pointer if `XXX` is not present. Then use sites that know it will be present can just dereference the helper call directly and other sites can do `if (auto* xxx = y.getXXX()) { ... }`. There is some danger of dangling pointers with this pattern when `y` is temporary, though.

@kripken, do you have a strong preference about what pattern to use here?

Reply:
I don't feel strongly. Returning null seems consistent with other stuff, so I'd lean slightly to that.